components/profile-controller/controllers/profile_controller.go

```diff
@@ -616,15 +624,59 @@ func removeString(slice []string, s string) (result []string) {
 	return
 }
 
+func mergeLabelPropertiesToMap(logger logr.Logger) {
+	file, err := os.Open(NAMESPACE_LABELS_PROPERTIES)
+	if err != nil {
+		logger.Info("namespace labels properties file doesn't exist, using default value")
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		label := scanner.Text()
+		arr := strings.Split(label, "=")
+		if len(arr) > 2 || len(arr) == 0 {
+			logger.Info("label config format incorrect, should be {key}={value}", "label", label)
+		} else {
+			if arr[0] == "" {
+				logger.Info("label key should not be empty", "label", label)
+			} else if len(arr) == 2 {
+				// Add or overwrite label map if value exists.
+				kubeflowNamespaceLabels[arr[0]] = arr[1]
+			} else {
+				// Set value to be empty if nothing exists after "=".
+				// It means this label needs to be removed.
+				kubeflowNamespaceLabels[arr[0]] = ""
+			}
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		logger.Error(err, "Error reading namespace labels properties file")
+	}
+}
+
 func updateNamespaceLabels(ns *corev1.Namespace) bool {
 	updated := false
+
 	if ns.Labels == nil {
 		ns.Labels = make(map[string]string)
 	}
+
 	for k, v := range kubeflowNamespaceLabels {
-		if _, ok := ns.Labels[k]; !ok {
-			ns.Labels[k] = v
-			updated = true
+		_, ok := ns.Labels[k]
+		if len(v) == 0 {
+			// When there is an empty value, k should be removed.
+			if ok {
+				delete(ns.Labels, k)
+				updated = true
+			}
+		} else {
+			if !ok {
```

Comment:
For discussion,

Not sure what would be our intended behavior, if we start profile controller with default labels, and later decided to configure custom labels. Existing namespaces won't be updated to the new labels, because the labels already exist, so we skip the update process.

Reply:
See the same comment from above, which is to **enforce** labels from VolumeMount, and only **add** label from `kubeflowNamespaceLabels` if label doesn't exist. 


> I have thought about it: So the current `kubeflowNamespaceLabels` will not overwrite any existing labels, but our new feature is going to enforce addition/removal/setting of a label. So I created another layer to `enforce` labels received from VolumeMount, and keep the `kubeflowNamespaceLabels` intact.